contracts/utils/safemath.cairo

```diff
@@ -48,7 +50,53 @@ func uint256_checked_sub_lt{
     uint256_check(b)
 
     let (is_lt) = uint256_lt(b, a)
-    assert_not_zero(is_lt)
+    assert is_lt = TRUE
     let (c: Uint256) = uint256_sub(a, b)
     return (c)
 end
+
+# Multiplies two integers.
+# Reverts if product is greater than 2^256.
+func uint256_checked_mul{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    } (a: Uint256, b: Uint256) -> (c: Uint256):
+    alloc_locals
+    uint256_check(a)
+    uint256_check(b)
+    let (is_zero) = uint256_eq(a, Uint256(0, 0))
+    if is_zero == TRUE:
+        return (a)
+    end
```

Comment:
checking only `a` is a performance improvement in Solidity, but is it the case here too? it'd be good to benchmark with just `a`, both, and none of them.

Reply:
Great catch. See below